core/src/main/java/org/renjin/stats/internals/distributions/RNG.java

```diff
@@ -142,28 +163,36 @@ public static DoubleVector rexp(@Current Context context, int n, double invrate)
   }
 
   @Internal
-  public static DoubleVector rpois(@Current Context context, int n, double mu) {
+  public static DoubleVector rpois(@Current Context context, Vector n, double mu) {
+    int n1 = (n.length() != 1) ? n.length() : n.getElementAsInt(0);
     DoubleArrayVector.Builder vb = new DoubleArrayVector.Builder();
-    for (int i = 0; i < n; i++) {
+    for (int i = 0; i < n1; i++) {
       vb.add(Poisson.rpois(context.getSession(), mu));
     }
     return (vb.build());
   }
 
   @Internal
-  public static DoubleVector rsignrank(@Current Context context, int nn, double n) {
+  public static DoubleVector rsignrank(@Current Context context, Vector nn, double n) {
+    int nn1 = (nn.length() != 1) ? nn.length() : nn.getElementAsInt(0);
```

Comment:
Can you extract this to a function so it's the  same for all the functions. Try to avoid distinguishing variable names by number. If the difference between the two variables is that one is a vector and the other is not, then `nVector` and `n` might be logical choices